### PR TITLE
PI-622 e2e tests for custody-key-dates-and-delius - temporary version

### PR DESCRIPTION
**Pull request by anthony-britton-moj**

### CODE REVIEW

File: steps/delius/event/find-events.ts [hunk 1]
```diff
@@ -29,3 +31,22 @@ export async function isInEventContext(page: Page, crn: string, eventNumber: num
...
+    await refreshUntil(page, () => verifyDate(page, 'Sentence Expiry Date:', date))
+    await verifyDate(page, 'Sentence Expiry Date:', date)
+    await verifyDate(page, 'Licence Expiry Date:', date)
+    await verifyTableDate(page, 'Parole Eligibility Date', date)
+    await verifyTableDate(page, 'Auto-Conditional Release Date', date)
+}
+
+const verifyDate = async (page: Page, label: string, date: Date) => {
+    const id = await page.locator('label', { hasText: label }).getAttribute('for')
+    await expect(page.locator(`[id="${id}"]`)).toHaveText(DeliusDateFormatter(date))
```
Comment (marcus-bcl) [inline steps/delius/event/find-events.ts:46]: Playwright recently added a method for this:
```suggestion
    await expect(page.getByLabel(label)).toHaveText(DeliusDateFormatter(date))
```
See https://playwright.dev/docs/input#text-input

File: package.json [hunk 2]
```diff
@@ -27,6 +28,7 @@
   },
   "dependencies": {
     "@types/date-fns": "^2.6.0",
-    "date-fns": "^2.29.3"
+    "date-fns": "^2.29.3",
+    "uuid": "^9.0.0"
```
Comment (marcus-bcl) [inline package.json:30]: We don't need to do it now, but we should move all these to the dev-dependencies for consistency (everything in this repo is dev)

File: test-data/environments/test.ts [hunk 3]
```diff
@@ -42,4 +42,8 @@ export const testEnvironmentData: TestData = {
             providerName: 'NPS North East',
         },
     },
+    keyDate: {
```
Comment (marcus-bcl) [inline test-data/environments/test.ts]: Maybe something like 'sentencedPrisoner' would be a more descriptive name?

File: tests/custody-key-dates-and-delius/custody-key-dates.spec.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import { verifyContacts } from '../../steps/delius/contact/find-contacts.js'
+import { contact } from '../../steps/delius/utils/contact.js'
+import { data } from '../../test-data/test-data.js'
+import { commonData } from '../../test-data/environments/common.js'
+
+test('Update Custody Key Dates', async ({ page }) => {
+    await deliusLogin(page)
+
+    const today = new Date()
+    today.setFullYear(today.getFullYear() + 2)
```
Comment (marcus-bcl) [inline tests/custody-key-dates-and-delius/custody-key-dates.spec.ts]: Will this only change the date once a day, or does it handle re-runs?

**Review by marcus-bcl — COMMENTED** [review]
Few minor comments

File: steps/delius/event/find-events.ts (hunk 1 above)
Comment (anthony-britton-moj) [reply steps/delius/event/find-events.ts:46]: I'll give it a go, but not sure that will work, as that is for getting an input the label is assigned to. This is a span we are reading from.

File: tests/custody-key-dates-and-delius/custody-key-dates.spec.ts (hunk 4 above)
Comment (anthony-britton-moj) [reply tests/custody-key-dates-and-delius/custody-key-dates.spec.ts]: resolved now

File: steps/delius/event/find-events.ts (hunk 1 above)
Comment (anthony-britton-moj) [reply steps/delius/event/find-events.ts:46]: tried it and it doesn't work for reading those spans
` 
   * Allows locating input elements by the text of the associated label. For example, this method will find the input by
   `

Comment (marcus-bcl) [reply steps/delius/event/find-events.ts:46]: Ah okay, my bad!

File: package.json (hunk 2 above)
Comment (anthony-britton-moj) [reply package.json:30]: done

**Review by marcus-bcl — APPROVED** [review]

**Pull request merged**